calculator/parser.py

```diff
@@ -56,16 +66,49 @@ def push(e, stack):
     stack.append(e)
 
 
-def pop(stack):
+def pop(list, type):
     """
-    Retrieves the last element from the given stack and deletes it, which mutates it
+    Retrieves the last element from the given stack or queue and deletes it, which mutates it
 
     Parameters
     ----------
-    stack : list
-        The stack (in list form) to be operated on
+    list : list
+        The list to be operated on
+
+    type : string
+        Whether or not it is a queue or a stack
+
     """
-    return stack.pop()
+    if type == "s":
+        return list.pop()
+
+    elif type == "q":
+        return list.pop(0)
+
+
+def peek(list, type):
+    """
+    Returns the top value of the given stack
+
+    Parameter
+    ---------
+    list : list
+        The stack or queue (in list form) to operated on
+
+    Returns
+    -------
+    token : The top token of the stack or queue
+    """
+
+    try:
+        if type == "s":
+            return list[-1]
+
+        elif type == "q":
+            return list[0]
+
+    except:
+        return None
 
 
 def is_numeric_token(token):
```

Comment:
I'd recommend reimplementing `is_numeric_token` as:
```
import numerics
...
def is_numeric_token(token):
    return isinstance(token, numbers.Number)
```

Reply:
ok 👍 , btw, it is `numbers` not `numerics`

Reply:
👍